raft.go

```diff
@@ -627,33 +630,41 @@ func (r *Raft) leaderLoop() {
 				}
 			}
 
-			var numProcessed int
 			start := time.Now()
+			var groupReady []*list.Element
+			var groupFutures = make(map[uint64]*logFuture)
+			var lastInGroup uint64
```

Comment:
nit: I'd name this `lastIdxInGroup` to be obvious that it's an index not a pointer to an actual future or leg entry. Not a big deal though as it's only used locally here.